geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java

```diff
@@ -3799,12 +3799,18 @@ public RemovedEventInfo(Long counter, String regionName, Object key) {
 
     private static volatile ExpirationAttributes testExpAtts;
 
+    private final int expTime;
+
+    ThreadIdentifierCustomExpiry() {
+      expTime = calculateThreadIdExpiryTime();
+    }
+
     @Override
     public ExpirationAttributes getExpiry(Region.Entry entry) {
       // Use key to determine expiration.
       Object key = entry.getKey();
       if (key instanceof ThreadIdentifier) {
-        final int expTime = calculateThreadIdExpiryTime();
+        // TODO: inject subclass of ThreadIdentifierCustomExpiry and move next block to test class
         if (expTime != DEFAULT_THREAD_ID_EXPIRY_TIME) {
```

Comment:
I hoisted "expTime = calculateThreadIdExpiryTime();" from here to the constructor.